test_runner/regress/test_logging.py

```diff
@@ -0,0 +1,49 @@
+import uuid
+
+import pytest
+from fixtures.log_helper import log
+from fixtures.neon_fixtures import NeonEnvBuilder
+from fixtures.utils import wait_until
+
+
+@pytest.mark.parametrize("level", ["trace", "debug", "info", "warn", "error"])
+def test_logging_event_count(neon_env_builder: NeonEnvBuilder, level: str):
+    # self-test: make sure the event is logged (i.e., our testing endpoint works)
+    log_expected = {
+        "trace": False,
+        "debug": False,
+        "info": True,
+        "warn": True,
+        "error": True,
+    }[level]
+
+    env = neon_env_builder.init_start()
+    ps_http = env.pageserver.http_client()
+    msg_id = uuid.uuid4().hex
+
+    # NB: the _total suffix is added by our prometheus client
+    before = ps_http.get_metric_value("libmetrics_tracing_event_count_total", {"level": level})
+
+    # post the event
+    ps_http.post_tracing_event(level, msg_id)
+    if log_expected:
+        env.pageserver.allowed_errors.append(f".*{msg_id}.*")
+
+    def assert_logged():
+        if not log_expected:
+            return
+        assert env.pageserver.log_contains(f".*{msg_id}.*")
+
+    wait_until(10, 0.5, assert_logged)
+
+    # make sure it's counted
+    def assert_metric_value():
+        if not log_expected:
+            return
+        # NB: the _total suffix is added by our prometheus client
+        val = ps_http.get_metric_value("libmetrics_tracing_event_count_total", {"level": level})
+        val = val or 0.0
+        log.info("libmetrics_tracing_event_count: %s", val)
+        assert val > (before or 0.0)
+
+    wait_until(10, 1, assert_metric_value)
```

Comment:
What's the rationale for both tests?

Reply:
The rationale is to ensure that the metric works as expected.

The mere presence in `metrics.py` is not enough for that.

Reply:
Agreed, however this will be quite expensive test with parametrized different levels and we already have a rust test for this, so this basically tests that we have configured the layer.

I'd hope we could get away not doing these kinds of tests, maybe roll this up with some other test case where we just assert that there is non-zero amount of infos being logged? That is, if we really want to test that the layer gets configured.

Reply:
Expensive? I disagree.

<img width="912" alt="image" src="https://user-images.githubusercontent.com/956573/234241271-a5434db1-8bbb-4120-80ec-e66583449224.png">

Reply:
More lines of code is still something to be maintained, but you already wrote it so I guess we are fine with that. I don't see there being a particular risk managed by this python test (works + configured vs. rust test showing it works).

But lets resolve.